external/js/cothority/src/skipchain/skipchain-rpc.ts

```diff
@@ -104,68 +106,54 @@ export default class SkipchainRPC {
     async getAllSkipChainIDs(): Promise<Buffer[]> {
         const req = new GetAllSkipChainIDs();
 
-        const ret = await this.conn[0].send<GetAllSkipChainIDsReply>(req, GetAllSkipChainIDsReply);
+        const ret = await this.pool.send<GetAllSkipChainIDsReply>(req, GetAllSkipChainIDsReply);
 
         return ret.skipChainIDs.map((id) => Buffer.from(id));
     }
 
     /**
      * Get the shortest path to the more recent block starting from latestID
      *
-     * @param latestID ID of the block
+     * @param latestID  ID of the block
+     * @param verify    Verify the integrity of the chain when true
      * @returns a promise that resolves with the list of blocks
      */
-    async getUpdateChain(latestID: Buffer): Promise<SkipBlock[]> {
+    async getUpdateChain(latestID: Buffer, verify = true): Promise<SkipBlock[]> {
         const req = new GetUpdateChain({ latestID });
-        const ret = await this.conn[0].send<GetUpdateChainReply>(req, GetUpdateChainReply);
+        const ret = await this.pool.send<GetUpdateChainReply>(req, GetUpdateChainReply);
+        const blocks = ret.update;
 
-        const err = this.verifyChain(ret.update, latestID);
-        if (err) {
-            throw new Error(`invalid chain received: ${err.message}`);
+        const last = blocks[blocks.length - 1];
```

Comment:
what would happen if blocks is empty?

Reply:
You can check in your browser: in JS you get _undefined_ for undefined indices